styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java

```diff
@@ -62,20 +65,59 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
     final String operation = method.getName();
 
     final Instant t0 = time.get();
-    final Object ret;
-    try {
-      ret = method.invoke(delegate, args);
-    } catch (InvocationTargetException e) {
-      throw e.getTargetException();
-    }
+    final Try<?> result = Try.of(() -> {
+      try {
+        return method.invoke(delegate, args);
+      } catch (InvocationTargetException e) {
+        throw e.getTargetException();
+      }
+    });
+
     final long durationMillis = t0.until(time.get(), ChronoUnit.MILLIS);
 
+    // TODO: Instrument at a lower level to catch errors with greater granularity. At this level we might not see
+    //       all errors as the implementation might swallow exceptions, perform retries, etc.
+    result.onFailure(error -> {
+      if (delegate instanceof DockerRunner) {
```

Comment:
Move this to L90? We already check `instaceof` there. Or if we have a separated helper class handling reporting failure, we can pass `delegate` to that class and do all the dirty check there. So instead of `reportDockerOperationError` we can do `reportOperationError`.

Reply:
Yeah, I think I'd like to maybe work on moving this to `KubernetesDockerRunner`.